src/lsm/superblock.zig

```diff
@@ -0,0 +1,1491 @@
+const std = @import("std");
+const assert = std.debug.assert;
+const crypto = std.crypto;
+const mem = std.mem;
+const meta = std.meta;
+const os = std.os;
+
+const config = @import("../config.zig");
+const utils = @import("../utils.zig");
+const vsr = @import("../vsr.zig");
+
+const BlockFreeSet = @import("block_free_set.zig").BlockFreeSet;
+
+const log = std.log.scoped(.superblock);
+
+/// Identifies the type of a sector or block. Protects against misdirected I/O across valid types.
+pub const Magic = enum(u8) {
+    superblock,
+    manifest,
+    prepare,
+    index,
+    filter,
+    data,
```

Comment:
> Where are these (non-superblock) magic numbers used?

They're going to move out of SuperBlock into VSR.

> Right now the magic field is positioned after both checksum and copy in the SuperBlockSector. Will these other sector/block types also have an extra padding byte prior to the magic number, or should copy and magic swap positions?

We still need to introduce this `magic` field into our other stored data struct layouts. Where the `magic` field is in each of the structs can depend on the struct itself (e.g. how we pack for alignment), however it needs to be covered by the checksum.

On the other hand, the `copy` field is unique to the `SuperBlockSector` struct, and this is the one field not covered by the checksum (so that copies have the same checksum, and so that they can be referenced by the next superblock version in its `parent` field), which is why it comes directly after the checksum, so we can skip the whole uncovered prefix `{ checksum, copy }` in one shot without having to skip over fields.